### PR TITLE
docs: update android sdk

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary
1. add the screen view definition
2. update the definition of session
3. update the definition of user engagement
4. update the example event for record an event
5. add the data type for each attribute
6. add reference link
7. adjust the content structure
8. update the error code
9. fix typo

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/sdk-manual/android.md [hunk 1]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withTrackScreenViewEvents()     | boolean        | false    | true          | whether auto record screen view events                                                      |
+| withTrackUserEngagementEvents() | boolean        | false    | true          | whether auto record user engagement events in App                                           |
+| withTrackAppExceptionEvents()   | boolean        | false    | true          | whether auto record app exception events                                                    |
+| withLogEvents()                 | boolean        | false    | true          | whether to automatically print event json for debugging events, [Learn more](#debug-events) |
+| withCustomDns()                 | String         | false    | --            | the method for setting your custom dns, [Learn more](#configure-custom-dns)                 |
+| withCompressEvents()            | boolean        | false    | true          | whether to compress event content by gzip when uploading events.                            |
```
Comment (zxkane) [inline docs/en/sdk-manual/android.md:198]: remove the end dot(.) or add it for all

File: docs/en/sdk-manual/android.md [hunk 2]
```diff
@@ -75,9 +75,90 @@ public void onCreate() {
...
+```
+
+#### Add global attribute
+
+```java
+import software.aws.solution.clickstream.ClickstreamAttribute;
+import software.aws.solution.clickstream.ClickstreamAnalytics;
+
+ClickstreamAttribute globalAttribute = ClickstreamAttribute.builder()
+    .add("channel", "HUAWEI")
```
Comment (zxkane) [inline docs/en/sdk-manual/android.md:108]: `Google Play` is better

File: docs/en/sdk-manual/android.md [hunk 3]
```diff
@@ -145,172 +250,203 @@ Clickstream Android SDK supports the following data types:
...
-    }
+	"hashCode": "80452b0",
+	"unique_id": "c84ad28d-16a8-4af4-a331-f34cdc7a7a18",
+	"event_type": "add_to_cart",
+	"event_id": "460daa08-0717-4385-8f2e-acb5bd019ee7",
+	"timestamp": 1667877566697,
+	"device_id": "f24bec657ea8eff7",
+	"platform": "Android",
+	"os_version": "10",
+	"make": "HUAWEI",
```
Comment (zxkane) [inline docs/en/sdk-manual/android.md:344]: Sumang is better

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/android.md [hunk 4]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
 ```
 
-### 5. Record event
-
-Add the following code where you need to report an event. For more information, refer to [GitHub](https://github.com/awslabs/clickstream-android#start-using).
+Here is an explanation of each method.
+
+| Method name                     | Parameter type | Required | Default value | Description                                                                                 |
+|---------------------------------|----------------|----------|---------------|---------------------------------------------------------------------------------------------|
+| withAppId()                     | String         | true     | --            | the app id of your application in control plane                                             |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:188]: replaced all `control plane` with `web console`

File: docs/en/sdk-manual/android.md [hunk 5]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withEndpoint()                  | String         | true     | --            | the endpoint path you will upload the event to Clickstream ingestion server                 |
+| withAuthCookie()                | String         | false    | --            | your auth cookie for AWS application load balancer auth cookie                              |
+| withSendEventsInterval()        | long           | false    | 100000        | event sending interval millisecond                                                          |
+| withSessionTimeoutDuration()    | long           | false    | 5000          | the duration for session timeout milliseconds                                               |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:192]: fixed

File: docs/en/sdk-manual/android.md [hunk 6]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withTrackScreenViewEvents()     | boolean        | false    | true          | whether auto record screen view events                                                      |
+| withTrackUserEngagementEvents() | boolean        | false    | true          | whether auto record user engagement events in App                                           |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:194]: fixed

File: docs/en/sdk-manual/android.md [hunk 7]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withTrackAppExceptionEvents()   | boolean        | false    | true          | whether auto record app exception events                                                    |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:195]: fixed.

File: docs/en/sdk-manual/android.md [hunk 8]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+Here is an explanation of each method.
+
+| Method name                     | Parameter type | Required | Default value | Description                                                                                 |
+|---------------------------------|----------------|----------|---------------|---------------------------------------------------------------------------------------------|
+| withAppId()                     | String         | true     | --            | the app id of your application in control plane                                             |
+| withEndpoint()                  | String         | true     | --            | the endpoint path you will upload the event to Clickstream ingestion server                 |
+| withAuthCookie()                | String         | false    | --            | your auth cookie for AWS application load balancer auth cookie                              |
+| withSendEventsInterval()        | long           | false    | 100000        | event sending interval millisecond                                                          |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:191]: fixes

File: docs/en/sdk-manual/android.md [hunk 9]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withSessionTimeoutDuration()    | long           | false    | 5000          | the duration for session timeout milliseconds                                               |
+| withTrackScreenViewEvents()     | boolean        | false    | true          | whether auto record screen view events                                                      |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:193]: fixed

File: docs/en/sdk-manual/android.md [hunk 10]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withTrackUserEngagementEvents() | boolean        | false    | true          | whether auto record user engagement events in App                                           |
+| withTrackAppExceptionEvents()   | boolean        | false    | true          | whether auto record app exception events                                                    |
+| withLogEvents()                 | boolean        | false    | true          | whether to automatically print event json for debugging events, [Learn more](#debug-events) |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:196]: replaced

File: docs/en/sdk-manual/android.md [hunk 11]
```diff
@@ -87,43 +168,67 @@ import software.aws.solution.clickstream.ClickstreamAnalytics;
...
+| withCustomDns()                 | String         | false    | --            | the method for setting your custom dns, [Learn more](#configure-custom-dns)                 |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:197]: replaced

File: docs/en/sdk-manual/android.md (hunk 2 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:108]: changed

File: docs/en/sdk-manual/android.md (hunk 3 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:344]: changed to `Samsung`

File: docs/en/sdk-manual/android.md (hunk 1 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/android.md:198]: delete this dot

**Review by zxkane — APPROVED** [review]

**Pull request merged**